libnavigation-core/src/main/java/com/mapbox/navigation/core/reroute/MapboxRerouteController.kt

```diff
@@ -94,20 +99,36 @@ internal class MapboxRerouteController(
     }
 
     override fun reroute(routesCallback: RerouteController.RoutesCallback) {
-        reroute(
-            NavigationRerouteController.RoutesCallback { routes, _ ->
-                routesCallback.onNewRoutes(routes.toDirectionsRoutes())
-            }
-        )
+        reroute { routes, _ ->
+            routesCallback.onNewRoutes(routes.toDirectionsRoutes())
+        }
     }
 
     override fun reroute(callback: NavigationRerouteController.RoutesCallback) {
         interrupt()
         state = RerouteState.FetchingRoute
-        logD(
-            "Fetching route",
-            LOG_CATEGORY
-        )
+        logD("Fetching route", LOG_CATEGORY)
+
+        ifNonNull(
+            directionsSession.routes,
+            tripSession.getRouteProgress()?.routeAlternativeId,
+        ) { routes, routeAlternativeId ->
+            val relevantAlternative = routes.find { it.routeId == routeAlternativeId }
+            if (relevantAlternative != null) {
+                val newList = mutableListOf(relevantAlternative).apply {
+                    addAll(routes.toMutableList().also { it.removeFirstOrNull() })
```

Comment:
```suggestion
                    addAll(routes.toMutableList().drop(1))
```

Reply:
drop returns `List`, removeFirstOrNull works with `MutableList`. Addressed in https://github.com/mapbox/mapbox-navigation-android/pull/5645#discussion_r842150161